### PR TITLE
`Development`: Add JavaDocs to UtilServices and Factories [15]

### PR DESCRIPTION
**Pull request by DominikRemo**
#### General
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://ls1intum.github.io/Artemis/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://ls1intum.github.io/Artemis/dev/development-process/#naming-conventions-for-github-pull-requests).

### Motivation and Context

Improve the readability of UtilServices and Factories by adding JavaDoc to the helper methods

### Description

Added documentation to following files:
- [x] util/PageableSearchUtilService.java
- [x] notification/NotificationFactory.java
- [x] plagiarism/PlagiarismUtilService.java

#### Code Review
- [x] Code Review 1
- [x] Code Review 2

### CODE REVIEW

File: src/test/java/de/tum/in/www1/artemis/notification/NotificationFactory.java [hunk 1]
```diff
@@ -29,6 +44,13 @@ public static GroupNotification generateGroupNotification(ZonedDateTime notifica
         return groupNotification;
     }
 
+    /**
+     * Generates a SystemNotification with the given arguments.
+     *
+     * @param notificationDate The notification date of the SystemNotification
+     * @param expireDate       The expiry date of the SystemNotification
```
Comment (egekurt123) [inline src/test/java/de/tum/in/www1/artemis/notification/NotificationFactory.java:51]: parameter is named expireDate (wrong, it should be either expiryDate or expirationDate) and explanation expiry date (correct in british english)

**Review by egekurt123 — APPROVED** [review]
Documentation text LGTM

File: src/test/java/de/tum/in/www1/artemis/notification/NotificationFactory.java (hunk 1 above)
Comment (DominikRemo) [reply src/test/java/de/tum/in/www1/artemis/notification/NotificationFactory.java]: Implemented feedback
Please note that changing attribute and method names in SystemNotification.java is out of scope for this PR

File: src/test/java/de/tum/in/www1/artemis/plagiarism/PlagiarismUtilService.java [hunk 2]
```diff
@@ -51,6 +51,14 @@ public class PlagiarismUtilService {
     @Autowired
     private UserUtilService userUtilService;
 
+    /**
+     * Creates and saves a Course and a TextExercise. Also creates and saves StudentParticipations with similar Submissions.
```
Comment (JanaNF) [inline src/test/java/de/tum/in/www1/artemis/plagiarism/PlagiarismUtilService.java:55]: I would add an "It" before "Also .." --> "It also ..."

File: src/test/java/de/tum/in/www1/artemis/plagiarism/PlagiarismUtilService.java [hunk 3]
```diff
@@ -51,6 +51,14 @@ public class PlagiarismUtilService {
...
+     *
+     * @param userPrefix            The prefix for the user logins
+     * @param similarSubmissionText The text that is submitted by each student
```
Comment (JanaNF) [inline src/test/java/de/tum/in/www1/artemis/plagiarism/PlagiarismUtilService.java:58]: I would change the sentence to "The text that each student submits"

File: src/test/java/de/tum/in/www1/artemis/plagiarism/PlagiarismUtilService.java [hunk 4]
```diff
@@ -84,6 +92,15 @@ public Course addCourseWithOneFinishedTextExerciseAndSimilarSubmissions(String u
         return course;
     }
 
+    /**
+     * Creates and saves a ModellingExercise. Also creates and saves StudentParticipations with similar Submissions.
+     *
+     * @param userPrefix             The prefix for the user logins
+     * @param similarSubmissionModel The model that is submitted by each student
```
Comment (JanaNF) [inline src/test/java/de/tum/in/www1/artemis/plagiarism/PlagiarismUtilService.java:99]: I would change the sentence to "The model that each student submits". Then you don't use passive voice.

**Review by JanaNF — CHANGES_REQUESTED** [review]
I added some comments. Please have a look at it.

**Review by LenaKahle — APPROVED** [review]
Looks good.

**Review by xHadie — APPROVED** [review]
LGTM!

**Pull request merged**